app/main/routes.py

```diff
@@ -77,8 +80,10 @@ def upload():
                 fund=Config.FUND_NAME,
             )
         else:
-            current_app.logger.info("Validation errors found during upload")
-
+            if Config.ENABLE_VALIDATION_LOGGING:
+                current_app.logger.info(f"Validation errors found during upload: {validation_errors}")
```

Comment:
Bit of a nitpick: the text "Validation errors found during upload:" will be repeated many times throughout the logs if it is per message, might be clutered. Could you log the full sentance once, then log each validation message like
`Validation error: {valdation_error}`

Reply:
We can also combine this if/else into a single line by doing:

```
current_app.logger.info("Validation error" + f": {validation_errors}" if Config.ENABLE_VALIDATION_LOGGING else "")
```

Reply:
When I've been testing with multiple errors, they are printed in separate objects inside a single list like this:

 [{'cell_index': 'G21', 'description': 'The cell is blank but is required.',
              'section': 'Projects Progress Summary', 'sheet': 'Programme Progress'},
             {'cell_index': 'H30 or K30', 'description': 'The cell is blank but is required.',
              'section': 'Project Details', 'sheet': 'Project Admin'},
              {'cell_index': 'C29',
              'description': 'You’ve not entered any risks for this project. You must enter at least 1.',
              'section': 'Project Risks - Project 2',
              'sheet': 'Risk Register'}]

So there is only one printed 'Validation errors found during upload' statement in total - unless I'm misunderstanding what you mean here

Reply:
My instinct would be to loop over that list and log a single line per validation error starting with "validation_error", because when we inevitably need to extract these errors from the logs we can loop over each line and look for lines that match the pattern. I'm sure there would also be a way to do the same with the log format you have here, I'm just not sure what that would look like

Reply:
So is your suggestion to have the logs print out like this:

Validation error: {'cell_index': 'G21', 'description': 'The cell is blank but is required.', 'section': 'Projects Progress Summary', 'sheet': 'Programme Progress'}

Validation error: {'cell_index': 'H30 or K30', 'description': 'The cell is blank but is required.',
'section': 'Project Details', 'sheet': 'Project Admin'}

Or split up the objects themselves into single lines like this:
Validation errors:
     cell_index: G21
     description: The cell is blank but is required
     section: Projects Progress Summary
     sheet: Programme Progress

Validation errors:
     cell_index: H30 
     description: The cell is blank but is required
     section: Project Details
     sheet: Project Admin

for each object? Or have I misinterpeted both of those scenarios?

Reply:
Hi, I was suggesting the first option. 

Validation error: {'cell_index': 'G21', 'description': 'The cell is blank but is required.', 'section': 'Projects Progress Summary', 'sheet': 'Programme Progress'}

Validation error: {'cell_index': 'H30 or K30', 'description': 'The cell is blank but is required.',
'section': 'Project Details', 'sheet': 'Project Admin'}

But there are benifits to your implementation because then errors would be grouped by request. I just think having one log per validation error will make it simpler to extract errors from the logs.